src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -741,32 +948,34 @@ void DoorShutter_Draw(Actor* thisx, PlayState* play) {
                     Matrix_RotateY(M_PI, MTXMODE_APPLY);
                 }
             } else if (this->doorType == SHUTTER_BOSS) {
-                gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(D_809982D4[this->unk_168]));
+                gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(sBossDoorTextures[this->bossDoorTexIndex]));
             }
             gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx, "../z_door_shutter.c", 2109),
                       G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-            gSPDisplayList(POLY_OPA_DISP++, sp70->a);
-            if (this->unk_170 != 0.0f && sp70->b != NULL) {
-                Matrix_Translate(0, sp70->c * (1.0f - this->unk_170), sp70->translateZ, MTXMODE_APPLY);
+            gSPDisplayList(POLY_OPA_DISP++, gfxInfo->doorDL);
+            if (this->barsClosed != 0.0f && gfxInfo->barsDL != NULL) {
+                Matrix_Translate(0, gfxInfo->barsOpenOffsetY * (1.0f - this->barsClosed), gfxInfo->barsOffsetZ,
+                                 MTXMODE_APPLY);
                 gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx, "../z_door_shutter.c", 2119),
                           G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-                gSPDisplayList(POLY_OPA_DISP++, sp70->b);
+                gSPDisplayList(POLY_OPA_DISP++, gfxInfo->barsDL);
             }
         }
 
-        if (this->unk_16E != 0) {
+        if (this->lockTimer != 0) {
             Matrix_Scale(0.01f, 0.01f, 0.025f, MTXMODE_APPLY);
-            Actor_DrawDoorLock(play, this->unk_16E,
-                               (this->doorType == SHUTTER_BOSS)
-                                   ? DOORLOCK_BOSS
-                                   : ((this->unk_16C == 6) ? DOORLOCK_NORMAL_SPIRIT : DOORLOCK_NORMAL));
+            Actor_DrawDoorLock(
+                play, this->lockTimer,
+                (this->doorType == SHUTTER_BOSS)
+                    ? DOORLOCK_BOSS
+                    : ((this->gfxType == DOORSHUTTER_GFX_SPIRIT_TEMPLE) ? DOORLOCK_NORMAL_SPIRIT : DOORLOCK_NORMAL));
         }
 
         CLOSE_DISPS(play->state.gfxCtx, "../z_door_shutter.c", 2135);
     }
 }
 
-void func_8099803C(PlayState* play, s16 y, s16 countdown, s16 camId) {
+void DoorShutter_AddQuake(PlayState* play, s16 y, s16 countdown, s16 camId) {
```

Comment:
Just want to note that this function is also named `DoorShutter_RequestQuakeAndRumble` in https://github.com/zeldaret/oot/pull/1402